### ISSUE TITLE
[BR]: Failed during configuration: Have not found any log file for sshd jail

### ISSUE THREAD

**Issue by Arne19** [fail2ban/fail2ban#3567]
First sorry if this issue is already known, I couldn't find it here so far.

### Environment:
freshly install Debian Bookworm on a remote server.

- Fail2Ban version  : v1.0.2
- OS, including release name/version : Latest Debian 12
- [ ] Fail2Ban installed via OS/distribution mechanisms
- [x] You have not applied any additional foreign patches to the codebase
- [ ] Some customizations were done to the configuration (provide details below is so)

### The issue:

Hi, on a just freshly installed Debian Bookworm I had this issue with fail2ban. Restarting it, rebooting or reinstallation didn't help. I had to manually change the backend from auto to systemd and then it worked.

#### Steps to reproduce
install debian bookworm, install ufw and fail2ban. Enable and start fail2ban.

#### Expected behavior
Fail2ban directly starts and filtering sshd.

#### Observed behavior
Fail2ban can't be started until manually change the backend from auto to systemd

#### Any customizations done to /etc/fail2ban/ configuration

No customization was done on fail2ban configs.

```
```

#### Relevant parts of /var/log/fail2ban.log file:

```
 fail2ban.configreader   [2192]: WARNING 'allowipv6' not defined in 'Definition'. Using default one: 'auto'
 fail2ban                [2192]: ERROR   Failed during configuration: Have not found any log file for sshd jail
fail2ban                [2192]: ERROR   Async configuration of server failed
```

#### Relevant lines from monitored log files:

```
```

**Comment by sebres** [comment]
Dup of #3292
See https://github.com/fail2ban/fail2ban/issues/3292#issuecomment-1142503461 for a solution.

**Comment by dee0xeed** [comment]
Recently had same issue and noticed that on deb-12 instance,
where `fail2ban` is failing to start (with default config), there is no `/var/log/auth.log`:

```
ls -l /var/log/auth.log
ls: cannot access '/var/log/auth.log': No such file or directory
```

Changing `backend` to `systemd` helped, yes, but why there is no `auth.log`?

Additional info.
On system, where it works without config changes:

```
fail2ban-client status sshd
Status for the jail: sshd
|- Filter
|  |- Currently failed: 0
|  |- Total failed:     336
|  `- File list:        /var/log/auth.log   <<<<<<<<<<<<<
```

On system where I changed `backend` to `systemd`:

```
# fail2ban-client status sshd
Status for the jail: sshd
|- Filter
|  |- Currently failed: 0
|  |- Total failed:     0
|  `- Journal matches:  _SYSTEMD_UNIT=sshd.service + _COMM=sshd  <<<<<<<<<<<<<<
```

**Comment by dee0xeed** [comment]
after installing `rsyslog` and restarting `sshd` `fail2ban` starts successfully with vanilla config.
so this looks looks some sort of misconfiguration.

**Comment by voltaiac** [comment]
Can confirm too on debian 12. Installing rsyslog and restarting sshd/fail2ban it worked instantly.

**Comment by sebres** [comment]
> Changing backend to systemd helped, yes, but why there is no auth.log?

It logs to systemd journal (and the change of backend to systemd forces the jail to monitor the journal instead of log-file).

> after installing rsyslog and restarting sshd fail2ban starts successfully with vanilla config.

rsyslog would duplicate the messages (by selector-based filtering e. g. configured facilities) to log-files.

> so this looks looks some sort of misconfiguration.

Yes and my answer above explaining it.
Those who can read have a clear advantage.

**Comment by lauhub** [comment]
> Those who can read have a clear advantage.

@sebres I feel this sentence is condescending... I understand that you spent a lot of time because of this problem, but @dee0xeed is trying to help by spending time to share what he/shee found about this issue. 

And finding the correct workaround is not so easy because of the amount of messages in issue 3292, obviously.

So the **workaround is just at the bottom of this comment**.

## BUT

It appears that the `sshd_backend` value is already defined in `path-commons.conf` which is actually included from `paths-debian.conf`

So it **should** work on Debian, but it does not. If I compare with other distros' conf files, these files explicitly redefine the `sshd_backend` value.

As it does not work directly, may I suggest that the `paths-debian.conf` from `fail2ban` repos is modified to be consistent with the other ones.

This should include the following values that will probably be missing too in some specific environments:

> syslog_backend = systemd
> sshd_backend = systemd
> dropbear_backend = systemd
> proftpd_backend = systemd
> pureftpd_backend = systemd
> wuftpd_backend = systemd
> postfix_backend = systemd
> dovecot_backend = systemd

## TL;DR:

Add the following line to the `[DEFAULT]` section in `/etc/fail2ban/paths-debian.conf` (or in `jail.local`>`[DEFAULT]`):

```
sshd_backend = systemd
```

**Comment by sebres** [comment]
> is trying to help by spending time to share what he/shee found about this issue.
> and finding the correct workaround is not so easy because of the amount of messages in issue 3292, obviously.

Unless this issue grows in the same manner. Don't you see the contradiction in both statements?

The description and a "valid" workaround is directly in this https://github.com/fail2ban/fail2ban/issues/3292#issuecomment-1142503461.
The rest of the issue was rather the discussion about how we'll solve it regularly and what exactly must be fixed. And how one can prevent similar issues in the future.

And basically `backend`, like `logpath` or some other parameter could be considered as individual settings and sometimes (even in cases where maintainer are a bit too late) must be specified directly in the jail section. Emphasis is on **in the jail** and not somewhere else.

With more obscure workarounds and half-ready "solutions" the people may get more confused or even another issues.

For example, the setting of `sshd_backend = systemd` may work, but:
- it would not affect people, which distros or maintained fail2ban version rewrote it in `[sshd]` section (or included jail)... like once in debian, IIRC, where `sshd_backend` was overwritten directly in `jail.d/sshd.conf`.
- it could affect people, which used this substitution variable in another jails (or maintainer or whoever specified it for another jails);
- it would not work (and will be wrong), if sshd logging not to journal but to different file, so the `backend` needs to remain `auto`, but instead one has to set `logpath`. 

So all aforementioned people may be confused by such **"workaround at the bottom of comment."**
It is always important to understand what one does and why.

**Comment by pcwearingmedia** [comment]
> after installing `rsyslog` and restarting `sshd` `fail2ban` starts successfully with vanilla config. so this looks looks some sort of misconfiguration.

Yep, this works. Also running: Debian GNU/Linux 12 (bookworm)

**Comment by sebres** [comment]
See https://github.com/fail2ban/fail2ban/issues/3567#issuecomment-2602132615

The installing of rsyslog is more or less dirty workaround - it would simply begin to log to /var/log/auth.log (or whatever logfile) additionally to systemd-journal, so fail2ban find the file with file-related `backend`.
But after installing that, one needs to take care of log-rotation, etc.

I really don't understand, why the jail setting, suggested by me:
```ini
[sshd]
backend = system
```
is more complex than "installing rsyslog and restarting sshd", so that so many people following the advise with rsyslog instead?

**Comment by toonarmy14** [comment]
> ## TL;DR:
> 
> Add the following line to the [DEFAULT] section in /etc/fail2ban/paths-debian.conf (or in jail.local>[DEFAULT]):
> 
> `sshd_backend = systemd`
>  

While it may be safe/not cause any immediate issues, making @lauhub's [proposed solution](https://github.com/fail2ban/fail2ban/issues/3567#issuecomment-2625720591) in `paths-debian.local` will also resolve the issue, without needing to modify the `paths-debian.conf` file directly. 

Confirm working in Debian 12 (bookworm) inside unprivileged LXC on Proxmox.

```sh
# /etc/fail2ban/paths-debian.local
[DEFAULT]
sshd_backend = systemd

```

**Comment by sproggit** [comment]
Just wanted to mention in passing that I had the same issue on a Raspberry Pi 5 running Debian/Bookworm. 

Found my way here and applied the suggested edit to /etc/fail2ban/jail.local  and the service started perfectly at the next attempt. 

Sincere thanks to all who contributed to this fix - very much appreciated.
